src/frontend/mosh-server.cc

```diff
@@ -657,6 +664,12 @@ static void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &
   socklen_t saved_addr_len = 0;
   #endif
 
+  #ifdef HAVE_SYSLOG
+  Addr syslog_saved_addr;
```

Comment:
I think you can reuse the existing `saved_addr` and `saved_addr_len` variables, rather than duplicating them, even with the complication of `connected_utmp`.

Reply:
You can't because on address change in the connected_utmp case, the new addresses will be populated into saved_* before the utmp block ends meaning they will be set to the current connection address by the time we get to the syslog block.

Reply:
So, refactor the blocks :)